config/flow_counters.py

```diff
@@ -0,0 +1,154 @@
+from flow_counter_util.route import build_route_pattern, extract_route_pattern, check_route_flow_counter_support
+from utilities_common.cli import AbbreviationGroup, pass_db
+from utilities_common import cli # To make mock work in unit test
+
+#
+# 'flowcnt-route' group ('config flowcnt-route ...')
+#
+
+
+@click.group(cls=AbbreviationGroup, invoke_without_command=False)
+def flowcnt_route():
+    """Route flow counter related configuration tasks"""
+    pass
+
+
+@flowcnt_route.group()
+def pattern():
+    """Set pattern for route flow counter"""
+    pass
+
+
+@pattern.command(name='add')
+@click.option('-y', '--yes', is_flag=True)
+@click.option('--vrf', help='VRF/VNET name or default VRF')
+@click.option('--max', 'max_allowed_match', type=click.IntRange(1, 50), default=30, show_default=True, help='Max allowed match count')
+@click.argument('prefix-pattern', required=True)
+@pass_db
+def pattern_add(db, yes, vrf, max_allowed_match, prefix_pattern):
+    """Add pattern for route flow counter"""
+    _update_route_flow_counter_config(db, vrf, max_allowed_match, prefix_pattern, True, yes)
+
+
+@pattern.command(name='remove')
+@click.option('--vrf', help='VRF/VNET name or default VRF')
+@click.argument('prefix-pattern', required=True)
+@pass_db
+def pattern_remove(db, vrf, prefix_pattern):
+    """Remove pattern for route flow counter"""
+    _update_route_flow_counter_config(db, vrf, None, prefix_pattern, False)
+
+
+def _update_route_flow_counter_config(db, vrf, max_allowed_match, prefix_pattern, add, yes=False):
+    """
+    Update route flow counter config
+    :param db: db object
+    :param vrf: vrf string, empty vrf will be treated as default vrf
+    :param max_allowed_match: max allowed match count, $FLOW_COUNTER_ROUTE_MAX_MATCH_FIELD will be used if not specified
+    :param prefix_pattern: route prefix pattern, automatically add prefix length if not specified
+    :param add: True to add/set the configuration, otherwise remove
+    :param yes: Don't ask question if True
+    :return:
+    """
+    check_route_flow_counter_support()
+
+    if add:
+        try:
+            net = ipaddress.ip_network(prefix_pattern, strict=False)
+        except ValueError as e:
+            click.echo('Invalid prefix pattern: {}'.format(prefix_pattern))
+            exit(1)
+
+        if '/' not in prefix_pattern:
+            prefix_pattern += '/' + str(net.prefixlen)
+
+        key = build_route_pattern(vrf, prefix_pattern)
+        for _, cfgdb in db.cfgdb_clients.items():
+            if _try_find_existing_pattern(cfgdb, net, key, yes):
+                entry_data = cfgdb.get_entry(FLOW_COUNTER_ROUTE_PATTERN_TABLE, key)
+                old_max_allowed_match = entry_data.get(FLOW_COUNTER_ROUTE_MAX_MATCH_FIELD)
+                if old_max_allowed_match is not None and int(old_max_allowed_match) == max_allowed_match:
+                    click.echo('The route pattern is already configured, no configuration changed')
```

Comment:
maybe can rephrase like: 'The route pattern already exists, nothing to be changed.'